modules/percolator/src/main/java/org/elasticsearch/percolator/PercolatorFieldMapper.java

```diff
@@ -163,7 +171,24 @@ static RangeFieldMapper createExtractedRangeFieldBuilder(String name, RangeType
 
         @Override
         public Builder parse(String name, Map<String, Object> node, ParserContext parserContext) throws MapperParsingException {
-            return new Builder(name, parserContext.queryShardContextSupplier());
+            Builder builder = new Builder(name, parserContext.queryShardContextSupplier());
+            for (Iterator<Map.Entry<String, Object>> iterator = node.entrySet().iterator(); iterator.hasNext();) {
+                Map.Entry<String, Object> entry = iterator.next();
+                String propName = entry.getKey();
+                Object propNode = entry.getValue();
+                if (propName.contains("boost_fields")) {
+                    if (isObject(propNode)) {
+                        for (Map.Entry<?, ?> innerEntry : ((Map<?, ?>) propNode).entrySet()) {
+                            String fieldName = nodeStringValue(innerEntry.getKey(), null);
+                            builder.addBoostField(fieldName, nodeIntegerValue(innerEntry.getValue()));
```

Comment:
I initially used floats, but then realized that int sufficed here. But I agree with your consistency argument.